include/aws/s3/s3_client.h

```diff
@@ -495,17 +495,22 @@ void aws_s3_meta_request_cancel(struct aws_s3_meta_request *meta_request);
  * Note: pause is currently only supported on upload requests.
  * In order to pause an ongoing upload, call aws_s3_meta_request_pause() that
  * will return resume token. Token can be used to query the state of operation
- * at the pausing time. Use aws_byte_buf_init_from_s3_resume_token to serialize token.
- * To resume an upload that was paused, supply the serialized resume token in the meta
+ * at the pausing time.
+ * To resume an upload that was paused, supply resume token in the meta
  * request options structure member aws_s3_meta_request_options.resume_token.
  * The upload can be resumed either from the same client or a different one.
- * Resume token will be set to null in case of failures.
+ * Corner cases for resume upload are as follows:
+ * - upload is not MPU - fail with pause not supported error
+ * - pausing before MPU is created - NULL resume token returned. NULL resume
+ *   token is equivalent to restarting upload
+ * - pausing in the middle of part transfer - return resume token. scheduling of
+ *   new part uploads stops.
```

Comment:
I'm curious about where work resumes in the common case that we're in the middle of uploading...

Say an upload has 4 parts, and at the moment the user calls pause(), the state is:
- part 1 has been completely uploaded
- part 2 is in the middle of uploading
- part 3 is scheduled, but hasn't sent any network traffic yet
- part 4 hasn't been touched

... when the user uses that resume token, will that next upload send:
- just part 4 (because the last session already scheduled 1,2,3)
- OR: parts 2, 3, 4 (because we re-send any parts we're not 100% sure completed)
- or something else...

whichever it is, make a comment to the reader to dispel any questions they might have.
Something like "On resuming, partially uploaded parts will be re-sent"
or if it doesn't work that way "You must want for the current upload operation to complete before using the resume token"

oh yeah, DO they need to avoid overlapping a pause with a resume? If we're re-uploading parts that were mid-upload when we paused ... we might actually have two overlapping requests for the same exact part !?!?

Reply:
we talked
when resuming, the server tells us what parts it's already received. nice.